src/CompiledSouffle.h

```diff
@@ -159,4 +159,86 @@ class RelationWrapper : public Relation {
         return symTable;
     }
 };
+
+/** Nullary relations */
+class t_nullaries {
+private:
+    bool data;
+
+public:
+    t_nullaries() : data(false) {}
+    typedef ram::Tuple<RamDomain, 0> t_tuple;
+    struct context {};
+    context createContext() {
+        return context();
+    }
+    class iterator : public std::iterator<std::forward_iterator_tag, RamDomain*> {
+        bool value;
+
+    public:
+        iterator(bool v = false) : value(v) {}
+
+        const RamDomain* operator*() {
+            return nullptr;
+        }
+
+        bool operator==(const iterator& other) const {
+            return other.value == value;
+        }
+
+        bool operator!=(const iterator& other) const {
+            return other.value != value;
+        }
+
+        iterator& operator++() {
+            if (value) {
+                value = false;
+            }
+            return *this;
+        }
+    };
+    iterator begin() const {
+        return iterator(data);
+    }
+    iterator end() const {
+        return iterator();
+    }
+    void insert(const t_tuple& t) {
+        data = true;
+    }
+    void insert(const t_tuple& t, context& h) {
```

Comment:
`t` is a least consistent with the existing code, and is an abbreviation. `h` is not, so could be more informative.

It's also unused, so could be replaced with, say, `context& /* ctxt */` The same applies to other unused parameters in this class.